src/TFMiniPlus.cpp

```diff
@@ -11,72 +11,70 @@
  */
 
 #include "TFMiniPlus.h"
+#include "TFMiniPlusConstants.h"
 
-#define DATA_FRAME_MARKER 0x59
-#define DATA_FRAME_LENGTH 9
-#define CMD_FRAME_MARKER 0x5A
-#define MAX_CMD_RESPONSE_LENGTH 8
+template <std::size_t N>
```

Comment:
i tried like a peasant to just get the length from the command frames and init the array with that size ... but it basically told me to fuck off :D 
Neat trick nonetheless !

Reply:
People used to get statically allocated array lengths using sizeof(array)/sizeof(array[0]), which works but is not very safe. This is basically a safer version of that trick that uses templates to let the compiler verify at compile-time that the pointer passed is indeed to an array and that it's statically allocated.